plugins/modules/vmware_host_facts.py

```diff
@@ -253,8 +253,10 @@ def __init__(self, module):
             if len(self.host) > 1:
                 self.module.fail_json(msg="esxi_hostname matched multiple hosts")
             self.host = self.host[0]
+            self.esxi_time = None
         else:
             self.host = find_obj(self.content, [vim.HostSystem], None)
+            self.esxi_time = self.si.CurrentTime()
```

Comment:
Change looks good to me. Do you think we should use Ansible Setup format like this - 

```
"ansible_date_time": {
            "date": "2020-10-24",
            "day": "24",
            "epoch": "1603506912",
            "hour": "02",
            "iso8601": "2020-10-24T02:35:12Z",
            "iso8601_basic": "20201024T023512241794",
            "iso8601_basic_short": "20201024T023512",
            "iso8601_micro": "2020-10-24T02:35:12.241860Z",
            "minute": "35",
            "month": "10",
            "second": "12",
            "time": "02:35:12",
            "tz": "UTC",
            "tz_offset": "+0000",
            "weekday": "Saturday",
            "weekday_number": "6",
            "weeknumber": "42",
            "year": "2020"
        },
```

Reply:
Yeah, this structure contains much more specific info, but for us, the returned value `"host_current_time": "2020-11-30T08:43:56.472599+00:00"` is enough for now, I think we can get date and time by splitting this string. What do you think?

Reply:
@Akasurde Is this Ansible Setup format some kind of standard for ansible time information? I haven't seen this yet.

Anyway, is there a library function to create this? Something like `ansible_setup_format("2020-11-30T08:43:56.472599+00:00")` that returns the rather complex data structure you suggest?

@Tomorrow9 I don't think you should create this manually. But if ansible offers a standard way to create this from your timestamp it wouldn't hurt to use it, would it?

Reply:
@mariolenz yeah, if there is function like that would be helpful. I found this file to get the datetime structure as above,
https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/facts/system/date_time.py, but it seems not a standard interface to use it.

Reply:
There is no standard function to use. But we can refactor the collector code and make it reusable.

Reply:
I am also OK with writing the function in `vmware.py`

Reply:
There is also another collected info: `"ansible_bios_date": "2017-01-17T00:00:00+00:00"` in "ansible_facts". If we convert all datetime info into this, then how about user specified properties?

Reply:
I think we can keep the parent key as `host_date_time`.

Reply:
```
"host_date_time": {
            "date": "2020-10-24",
            "day": "24",
            "epoch": "1603506912",
            "hour": "02",
            "iso8601": "2020-10-24T02:35:12Z",
            "iso8601_basic": "20201024T023512241794",
            "iso8601_basic_short": "20201024T023512",
            "iso8601_micro": "2020-10-24T02:35:12.241860Z",
            "minute": "35",
            "month": "10",
            "second": "12",
            "time": "02:35:12",
            "tz": "UTC",
            "tz_offset": "+0000",
            "weekday": "Saturday",
            "weekday_number": "6",
            "weeknumber": "42",
            "year": "2020"
        },
```

Reply:
@Akasurde 
> I am also OK with writing the function in `vmware.py`

Nah, doesn't make sense to me. If there's a good reason to use this data structure for time information (and I _do_ see how this can be useful, especially if it's kind of a standard), ansible should provide a library option to create it from a time stamp. There's no need for all the collections out there to implement this for themselves. This:

> There is no standard function to use. But we can refactor the collector code and make it reusable.

sounds more like it ;-)

Reply:
If we refactor the collector code in https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/facts/system/date_time.py, to let it be a callable function, then this new added "host_date_time" info will depend on the Ansible release which containing that change.

Reply:
> If we refactor the collector code in https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/facts/system/date_time.py, to let it be a callable function, then this new added "host_date_time" info will depend on the Ansible release which containing that change.

Good point, haven't thought of this. So... copy the code over to `vmware.py` after all? Then other modules from this collection could use it, too. Could look like this:

```python
#timestamp is a datetime.datetime object
def ansible_date_time(timestamp):
  date_time_facts = {}
  
  utctimestamp = timestamp.astimezone(datetime.timezone.utc)
  
  date_time_facts['year'] = timestamp.strftime('%Y')
  date_time_facts['month'] = timestamp.strftime('%m')
  date_time_facts['weekday'] = timestamp.strftime('%A')
  date_time_facts['weekday_number'] = timestamp.strftime('%w')
  date_time_facts['weeknumber'] = timestamp.strftime('%W')
  date_time_facts['day'] = timestamp.strftime('%d')
  date_time_facts['hour'] = timestamp.strftime('%H')
  date_time_facts['minute'] = timestamp.strftime('%M')
  date_time_facts['second'] = timestamp.strftime('%S')
  date_time_facts['epoch'] = timestamp.strftime('%s')
  date_time_facts['date'] = timestamp.strftime('%Y-%m-%d')
  date_time_facts['time'] = timestamp.strftime('%H:%M:%S')
  date_time_facts['iso8601_micro'] = utctimestamp.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
  date_time_facts['iso8601'] = utctimestamp.strftime("%Y-%m-%dT%H:%M:%SZ")
  date_time_facts['iso8601_basic'] = timestamp.strftime("%Y%m%dT%H%M%S%f")
  date_time_facts['iso8601_basic_short'] = timestamp.strftime("%Y%m%dT%H%M%S")
  date_time_facts['tz'] = timestamp.strftime("%Z")
  date_time_facts['tz_offset'] = timestamp.strftime("%z")
  
  
  return date_time_facts
```

And you could use it like this:

```python
self.esxi_time = ansible_date_time(self.si.CurrentTime())
```

Reply:
Yeah, sure, I'll add this to vmware.py and use this function then. Thanks.

Reply:
@Akasurde @mariolenz I added a new function "ansible_date_time_facts" in vmware.py, now the output is:
```
"cluster": null,
            "host_date_time": {
                "date": "2020-12-03",
                "day": "03",
                "epoch": "1606988788",
                "hour": "09",
                "iso8601": "2020-12-03T09:46:28Z",
                "iso8601_basic": "20201203T094628938413",
                "iso8601_basic_short": "20201203T094628",
                "iso8601_micro": "2020-12-03T09:46:28.938413Z",
                "minute": "46",
                "month": "12",
                "second": "28",
                "time": "09:46:28",
                "tz": "UTC",
                "tz_offset": "+0000",
                "weekday": "Thursday",
                "weekday_number": "4",
                "weeknumber": "48",
                "year": "2020"
            },
```